File: Python/504.py
class Solution(object):
    def convertToBase7(self, num):
        if num == 0:
            return "0"
        b = "-" if num < 0 else ""
        a = ""
        num = abs(num)
        while num:
            a = str(num%7) + a
            num //= 7
        return b+a

File: Java/504.java
class Solution {
    public String convertToBase7(int num) {
        if(num == 0) return "0";
        String b = "",a = "";
        if(num < 0){
            b = "-";
            num *= -1;
        }
        while(num != 0){
            a = (char)(num%7 + '0') + a;
            num /= 7;
        }
        return b+a;
    }
}